GLPI 9.5 and marketplace compatibility

cconard96 opened the pull request:
Bump plugin and GLPI versions
Encapsulate JS to reduce global variable and function pollution
Removed hardcoded paths to plugin directory to allow support for marketplace
#70

xacobofg commented on setup.php:
```diff
@@ -78,6 +78,11 @@ function plugin_init_actualtime() {
       $PLUGIN_HOOKS['add_javascript']['actualtime'] = 'js/actualtime.js';
       $PLUGIN_HOOKS['item_purge']['actualtime']=['TicketTask'=>'plugin_actualtime_item_purge'];
 
+      if (strpos($_SERVER['REQUEST_URI'], 'ticket.form.php') !== false) {
+         $ajax_url = Plugin::getWebDir('actualtime') . '/ajax/timer.php';
+         echo Html::scriptBlock("document.addEventListener('DOMContentLoaded', function(){window.actualTime.init('$ajax_url');});");
+      }
+
```
With this code the active timer is only shown when you are on the ticket page, the normal operation is that it is shown on all pages

cconard96 commented:
> In ajax/timer.php you have to change the code with the new functions of js/actualtime.js

Ok. This will take a bit more work then. The issue before my last commit was that the script block was added to all front pages. When loading documents from document.send.php it added the script to all files and corrupted them.

cconard96 commented:
How is this? I fixed the function name and moved the JS init into a `pre_show_tab` hook function since the timer was shown with `post_show_tab`. I also fixed an issue with the new 'Unsaved changes' warning that was added in GLPI 9.5.

xacobofg commented on js/actualtime.js:
```diff
@@ -1,257 +1,265 @@
+         }
+      });
+   }
+}();
```
Hi @cconard96 thanks for your contribution.
I found that there is a global variable for javascript with the plugin paths `GLPI_PLUGINS_PATH`.
Adding this code at the end of the file will show the timer on all pages and should not affect the documents
`$(document).ready(function(){
var url = CFG_GLPI.root_doc+"/"+GLPI_PLUGINS_PATH.actualtime+"/ajax/timer.php";
window.actualTime.init(url);
});`

cconard96 replied:
Good catch. I guess I have some changes to make with my own plugins now too.

OscarBeiro commented:
Thanks @cconard96 !
